mods/blocks/fire.lua

```diff
@@ -0,0 +1,241 @@
+            name = "blocks_fire_" .. name .. "_animated.png",
+            animation = {type = "vertical_frames", aspect_w = 16, aspect_h = 16, length = 2.0}
+        }},
+        inventory_image = "blocks_fire_" .. name .. ".png",
+        drawtype = "firelike",
+        paramtype = "light",
+        sunlight_propagates = true,
+        walkable = false,
+        floodable = true,
+        light_source = default.LIGHT_MAX,
+        groups = {oddly_breakable_by_hand = 1},
+    })
+
+    minetest.register_node("blocks:embers_" .. name, {
+        description = S("Glowing " .. fire_description .. " Embers"),
+        tiles = {{
+            name = "blocks_embers_" .. name .. "_animated.png",
+            animation = {type = "vertical_frames", aspect_w = 16, aspect_h = 16, length = 2.0}
+        }},
+        paramtype = "light",
+        light_source = 6,
+        groups = {oddly_breakable_by_hand = 1},
+        sounds = default.node_sound_gravel_defaults()
+    })
+
+    minetest.register_node("blocks:campfire_" .. name, {
+        description = S(fire_description .. " Campfire"),
+        tiles = {{
+            name = "blocks_campfire_" .. name .. "_animated.png",
+            animation = {type = "vertical_frames", aspect_w = 32, aspect_h = 32, length = 2.0}
+        },
+        {
+            name = "blocks_firebranches_" .. name .. "_animated.png",
+            animation = {type = "vertical_frames", aspect_w = 16, aspect_h = 16, length = 2.0}
+        }},
+        inventory_image = "blocks_campfire_" .. name .. ".png",
+        drawtype = "mesh",
+        mesh = "campfire.obj",
+        paramtype = "light",
+        sunlight_propagates = true,
+        walkable = false,
+        floodable = true,
+        light_source = default.LIGHT_MAX,
+        groups = {oddly_breakable_by_hand = 1},
+        selection_box = {
+            type = "fixed",
+            fixed = {-14 / 16, -0.5, -14 / 16, 14 / 16, 16 / 16, 14 / 16},
+        },
+    })
+
+    minetest.register_node("blocks:pyre_" .. name, {
+        description = S("Fake " .. fire_description .. " Pyre"),
+        tiles = {{
+            name = "blocks_pyre_" .. name .. "_animated.png",
+            animation = {type = "vertical_frames", aspect_w = 64, aspect_h = 64, length = 2.0}
+        }},
+        inventory_image = "blocks_pyre_" .. name .. ".png",
+        drawtype = "mesh",
+        mesh = "pyre.obj",
+        paramtype = "light",
+        sunlight_propagates = true,
+        walkable = false,
+        floodable = true,
+        light_source = default.LIGHT_MAX,
+        groups = {oddly_breakable_by_hand = 1},
+        selection_box = {
+            type = "fixed",
+            fixed = {-20 / 16, -12 / 16, -20 / 16, 20 / 16, 48 / 16, 20 / 16},
+        },
+    })
+
+    -- Crafts
+
+    minetest.register_craft({
+        output = "blocks:fire_" .. name .. " 3",
+        recipe = {
+            {blocks.fire_types[i][3], blocks.fire_types[i][3], blocks.fire_types[i][3]},
```

Comment:
Can you store the value of this `blocks.fire_types[i][3]`, like you did with the 2 others?
It's a bit inefficient, since its fetching the value from scratch everytime.